### ISSUE TITLE
Kitty headless --single-instance

### ISSUE THREAD

**Issue by applesud** [kovidgoyal/kitty#3466]
I'm looking to improve the startup time of kitty on my desktop, as I open and close instances quite frequently. 
The `--single-instance` option makes a big difference, but it seems as though that single instance is closed when all of the windows are closed.

Could there be a way to run kitty as a headless single-instance (that could be launched on computer startup), similar to the Emacs `--daemon` option?

Sorry if this already exists; I couldn't see anything in the manual or docs.

**Comment by kovidgoyal** [comment]
No, I'm afraid not. Why does it matter if all instances get closed. All
that means is that *only the first* time you open a new window after
closing all windows it will be a bit slow (and we are talk order of
about 0.2 seconds here).

**Comment by dm17** [comment]
@kovidgoyal Figured I'd ask here... Can the --single-instance be specified in the kitty.conf on Linux so becomes default when launching kitty?

**Comment by kovidgoyal** [comment]
On Fri, May 19, 2023 at 12:29:50AM -0700, dm17 wrote:
> @kovidgoyal Figured I'd ask here... Can the --single-instance be specified in the kitty.conf on Linux so becomes default when launching kitty?

No, single instance must be specified at the command line, not in the
conf files.

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by AlaraBread** [comment]
would really like this feature, as its pretty common for me to open a terminal to do something quick and then close it again. it takes almost a full second on my computer to start kitty.

**Comment by AlaraBread** [comment]
> So put the following script named kitty in /usr/local/bin or somewhere else in your PATH ahead of where the actual kitty binary is:
> 
> ```
> #!/bin/sh
> exec /usr/bin/kitty --single-instance "$@"
> ```

this would only help with startup times if i already have another window of kitty open. opening the first window is still slow.

**Comment by kovidgoyal** [comment]
On Tue, Apr 22, 2025 at 05:14:28PM -0700, Alara Bread wrote:
> AlaraBread left a comment (kovidgoyal/kitty#3466)
>
> > So put the following script named kitty in /usr/local/bin or somewhere else in your PATH ahead of where the actual kitty binary is:
> >
> > ```
> > #!/bin/sh
> > exec /usr/bin/kitty --single-instance "$@"
> > ```
>
> this would only help with startup times if i already have another window of kitty open. opening the first window is still slow.

So keep a kitty window open in an unused workspace or minimized or
whatever such affordance your window manager gives you.

**Comment by kovidgoyal** [comment]
Oh and incidentally, in nightly, you can do:

```sh
kitty -1 --session <(echo "launch --allow-remote-control --hold kitten @ resize-os-window --action=hide")
```

this will run single instance with a hidden (unmapped) window. At least it works on wayland havent tested on X11.

**Comment by oxysoft** [comment]
Single instance is required for kitty windows to communicate and interop so from the user's perspective their system will break randomly when a kitty is inevitably opened through a non-standard route that hasn't had the argument applied and suddenly the user's plumbing doesn't kick in as expected. For example they may have a shortcut with a script to attach a kitty terminal pane into another terminal window using slop to select, and it will exclude windows inexplicably if they are not on the same process.
